neural_search/util.py

```diff
@@ -122,31 +123,51 @@ def recreate_collection(project_id: str, embedding_id: str) -> int:
     )
 
     ids, embeddings, payloads = zip(*all_object)
-    embeddings = np.array(embeddings)
     if len(embeddings) == 0:
         return status.HTTP_404_NOT_FOUND
-    vector_size = embeddings.shape[-1]
+    vector_size = 0
+    if len(embeddings) > 0 and embeddings[0] is not None:
+        vector_size = len(embeddings[0])
 
     qdrant_client.recreate_collection(
-        collection_name=embedding_id, vector_size=vector_size, distance="Euclid"
+        collection_name=embedding_id,
+        vectors_config=models.VectorParams(
+            size=vector_size, distance=models.Distance.EUCLID
+        ),
     )
-    all_none = all([payload is None for payload in payloads])
-    if all_none:
-        qdrant_client.upload_collection(
-            collection_name=embedding_id, vectors=embeddings, ids=ids
-        )
+    records = None
+    isPython = embedding.get(project_id, embedding_id).platform == "python"
+    if isPython:
```

Comment:
why is this stored in a variable & not the enum used?

```suggestion
    if embedding.get(project_id, embedding_id).platform == EmbeddingPlatform.PYTHON:
```